### DISCUSSION TITLE
Error: cannot sign payload without credential. Credential bytes are null

### DISCUSSION THREAD

**Discussion by kiknaio** [alchemyplatform/aa-sdk#1133] [Issues] [answered]
Most of the time when I pass authentication, and try to sign transactions, I'm getting this error
```
Error: cannot sign payload without credential. Credential bytes are null
```

Any idea what it's related to?

Seems like authentication related issue but I'm able to get signer, address, email, etc. Just signing the transactions are not working

**Comment by moldy530** [comment]
do you have a stack trace and example code?

**Reply by kiknaio** [reply]
Sure. 
<img width="490" alt="Screenshot 2024-11-08 at 13 11 58" src="https://github.com/user-attachments/assets/eca8842a-b967-4f23-91ed-7a7f7a01bed4">
Points to `sendUserOperationAsync` catch block.

**Reply by kiknaio** [reply]
<img width="879" alt="Screenshot 2024-11-08 at 13 53 27" src="https://github.com/user-attachments/assets/a41973c4-62bf-4fb6-a92f-6f8ca65cfbe7">
Also often I see this message and authentication is completely blocked

**Reply by kiknaio** [reply]
**Steps to Replicate the Issue:**

1. Log in using Passkey.
2. Refresh the page after successful login.
3. Attempt to send a transaction.

**Observed Result:**
The transaction fails, and an error message is displayed.

**Expected Behavior:**
The transaction should complete successfully without errors after a page refresh.

**Reply by kiknaio** [reply]
This is the error message I'm getting. Seems like it's coming from Turnkey
https://github.com/tkhq/frames/blob/53b177835ccf5c34719f71e40978fd9b2c7afe59/auth/index.html#L1163

**Reply by kiknaio** [reply]
One more thing I've found is that this specific error happens only after I sign in with Passkey.
In case of Email authentication, that's not an issue.

P.S.

I've checked Turnkey docs and I think that's expected behaviour. Which is extremely strange
https://github.com/tkhq/frames/blob/53b177835ccf5c34719f71e40978fd9b2c7afe59/auth/index.html#L1006-L1007

**Reply by moldy530** [reply]
Oh interesting... might be a few things going on here. 

For the oauth error @dphilipson might be able to help

**Reply by moldy530** [reply]
I'll try and debug as well the passkey one. I've seen passkey failures like this before too but can't repro consistently

**Reply by kiknaio** [reply]
These steps consistently reproduce the error:
* Sign in using Passkey.
* Sign a transaction.
* Refresh the page.
* Attempt to sign another transaction. It will fail.

**Reply by moldy530** [reply]
ok cool, will take a look at reproing this as well. thank you for the steps!

**Reply by moldy530** [reply]
good news! I've found the bug and should have a PR up today for it

**Reply by kiknaio** [reply]
> good news! I've found the bug and should have a PR up today for it

Amazing! Thanks

**Reply by moldy530** [reply, marked as answer]
PR is up https://github.com/alchemyplatform/aa-sdk/pull/1154, once approved I'll merge it and deploy the fix

**Reply by moldy530** [reply]
Should be deployed in the latest version of the sdk

**Reply by kiknaio** [reply]
Thanks! It's working perfectly now

**Reply by alexwaumann** [reply]
I believe this same issue is happening again. When I authenticate with a passkey and then call addOauthProvider on the signer (with popup, probably same with redirect), it seems to wipe out the passkey session because any action that requires the signer's sig results in the error `Error: cannot sign payload without credential. Credential bytes are null. Has a credential bundle been injected into the iframe?`. I am using v4.55.0.
